src/snowflake/snowpark/modin/plugin/_internal/resample_utils.py

```diff
@@ -478,39 +468,15 @@ def get_expected_resample_bins_frame(
     2020-01-07
     2020-01-09
     """
-    slice_width, slice_unit = rule_to_snowflake_width_and_slice_unit(rule)
-
-    index_column_snowflake_quoted_identifiers = (
-        generate_snowflake_quoted_identifiers_helper(
-            pandas_labels=[RESAMPLE_INDEX_LABEL]
-        )
-    )
-
-    # row_number ensures there are no gaps in the sequence.
-    all_resample_bins_col = dateadd(
-        slice_unit,
-        (row_number().over(Window.order_by(lit(1))) - 1) * slice_width,
-        to_timestamp_ntz(lit(start_date)),
-    ).as_(index_column_snowflake_quoted_identifiers[0])
-
-    rowcount = math.floor(
-        (native_pd.to_datetime(end_date) - native_pd.to_datetime(start_date))
-        / to_offset(rule)
-        + 1
-    )
-
-    expected_resample_bins_snowpark_frame = pd.session.generator(
-        all_resample_bins_col, rowcount=rowcount
-    )
-
+    expected_resample_bins_snowpark_frame = pd.date_range(
```

Comment:
Can you explain in the PR description (not necessarily as a code comment) why the new implementation is equivalent to the old one?